From: Divy Le Ray <divy@chelsio.com>
Date: Mon, 19 Jan 2009 06:01:32 +0000 (-0800)
Subject: cxgb3: Fix LRO misalignment
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fdavem%2Fnet-2.6.git;a=commitdiff_plain;h=eed087e367591fc08490d7c6c2779b4b72c8f20c

cxgb3: Fix LRO misalignment

The lro manager's frag_align_pad setting was missing,
leading to misaligned access to the skb passed up
to the stack.

Tested-by: Rick Jones <rick.jones2@hp.com>
Signed-off-by: Divy Le Ray <divy@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: John Jolly <jjolly@suse.de>
---

---
 drivers/net/cxgb3/sge.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/cxgb3/sge.c
+++ b/drivers/net/cxgb3/sge.c
@@ -2098,6 +2098,7 @@ static void init_lro_mgr(struct sge_qset
 {
 	lro_mgr->dev = qs->netdev;
 	lro_mgr->features = LRO_F_NAPI;
+	lro_mgr->frag_align_pad = NET_IP_ALIGN;
 	lro_mgr->ip_summed = CHECKSUM_UNNECESSARY;
 	lro_mgr->ip_summed_aggr = CHECKSUM_UNNECESSARY;
 	lro_mgr->max_desc = T3_MAX_LRO_SES;
